kyuubi-server/src/main/scala/org/apache/kyuubi/session/KyuubiSessionManager.scala

```diff
@@ -134,7 +134,8 @@ class KyuubiSessionManager private (name: String) extends SessionManager(name) {
       ipAddress: String,
       conf: Map[String, String],
       batchRequest: BatchRequest,
-      recoveryMetadata: Option[Metadata] = None): KyuubiBatchSessionImpl = {
+      recoveryMetadata: Option[Metadata] = None,
+      isResourceFromUpload: Boolean = false): KyuubiBatchSessionImpl = {
```

Comment:
with reserved key, you do not need change the constructor per time.

Reply:
Agree, it's a very good suggestion. Changed to setting a new reserved key in session configs instead.